lib/google_maps.ex

```diff
@@ -1311,6 +1311,69 @@ Each result contains the following fields:
     place_details({:place_id, place_id}, options)
   end
 
+
+  @doc """
+    A Timezone request returns timezone information for the given location.
+
+  ## Args:
+
+    * `location` - A comma-separated latitude / longitude tuple (eg, location = -33.86,151.20),
+      which represents the location to be searched.
+    * `timestamp` - Specifies the desired time in seconds after midnight, UTC,
+      from January 1, 1970. Google Maps Time Zone API uses timestamp to determine
+      if summer time should be applied. The hours before 1970 can be expressed as negative values.
+
+  ## Options:
+
+    * `language` — The language code, indicating in which language the
+      results should be returned, if possible. Searches are also biased
+      to the selected language; results in the selected language may be
+      given a higher ranking. See the [list of supported languages](https://developers.google.com/maps/faq#languagesupport)
+      and their codes. Note that we often update supported languages so
+      this list may not be exhaustive. If language is not supplied, the
+      Places service will attempt to use the native language of the
+      domain from which the request is sent.
+
+  ## Returns
+
+    This function returns `{:ok, body}` if the request is successful, and
+    Google returns data. The returned body is a map that contains four root
+    elements:
+
+    * `dstOffset` The time difference for summer time in seconds.
+      This value will be zero if the time zone is not in daylight saving time
+      during the specified timestamp.
+
+    * `rawOffset` The time difference with respect to UTC (in seconds)
+      for the determined location. This does not consider summer timetables.
+
+    * `timeZoneId` A string that contains the id. of "tz" in the time zone,
+      such as "United States / Los_Angeles" or "Australia / Sydney"
+
+    * `timeZoneName` A string that contains the name in long format the
+      time zone This field will be located if the parameter is configured of
+      language; p. eg, "Pacific Summer Time" or "Summer Time from Eastern Australia".
+
+    * `status` contains metadata on the request.
+
+  ## Examples
+
+      iex> {:ok, response} = GoogleMaps.timezone({8.6069305,104.7196242})
+      iex> is_map(response)
+      true
+
+      iex> {:ok, response} = GoogleMaps.timezone({8.6069305,104.7196242})
+      iex> response["timeZoneId"]
+      "Asia/Saigon"
+  """
+  @spec timezone(coordinate(), options()) :: Response.t()
+  def timezone(input, options \\ [])
+
+  def timezone({lat, lng}, options) when is_number(lat) and is_number(lng) do
```

Comment:
Would you mind adding another clause for when `location` is a comma-separate coordinate string? This is to fulfill the spec of `coordinate()` type.

Reply:
Would it be enough to just support the case when the first argument is a string? I don't see any validations in this module to ensure that that string is indeed `"lat, lon"`

Reply:
There is indeed no validation on that. I place the responsibility on the user to choose their input. The different clauses are there for flexibility only.

If somebody uses an invalid input, they will get an error from Google anyway.